client/src/components/leftPanel/LeftPanel.js

```diff
@@ -1,40 +1,84 @@
 import React, { Component } from 'react'
+import { withRouter } from 'react-router-dom'
 import _ from 'lodash'
-import vacuumLogo from '../../assets/vacuum-logo-light.svg'
 import moment from 'moment'
+
+import './LeftPanel.scss'
+import vacuumLogo from '../../assets/vacuum-logo-light.svg'
 import {parseTs} from '../../utils/helpers'
 import Tag from './Tag'
-import './LeftPanel.scss'
+import PortfoliosSelect from './PortfoliosSelect'
+
+
 
+function getPortfoliosFromSearchQuery(searchQuery) {
+  if (!searchQuery) return []
+  const splitSearchQuery = searchQuery.split('?portfolios=')
+  if (splitSearchQuery.length > 0) {
+    return splitSearchQuery[1].split(',').map((p) => ({label: p, value: p}))
+  }
+  return []
+} 
 class LeftPanel extends Component {
 
   state = {
     searchString: "",
+    portfolios: getPortfoliosFromSearchQuery(this.props.location.search),
     showArchived: false,
+    detailedView: true
   }
 
   onSearchChange = (event) => this.setState({ searchString: event.target.value })
   
-  onShowArchivedChange = (event) => this.setState({showArchived: event.target.checked})
+  onPortfoliosChange = (portfolios) => {
+    this.setState({ portfolios })
+    const searchQuery = portfolios.length > 0 ? `?portfolios=${encodeURI(portfolios.map(p => p.value))}` : ''
+    this.props.history.push(`${this.props.location.pathname}${searchQuery}`)
+  }
+
+  onShowArchivedChange = (event) => this.setState({ showArchived: event.target.checked })
+
+  toggleDetailedView = () => this.setState({ detailedView: !this.state.detailedView })
 
   render() {
-    const {tags} = this.props
+    const {tags, users, reports, portfolioOptions} = this.props
     for (const tag of tags) {
       tag.isLate =
         !tag.isArchived && moment().diff(parseTs(tag.lastTs), 'days') > tag.frequency
     }
     const partitionedTags = _.flatten(_.partition(tags, 'isLate'))
```

Comment:
Why do we partition something into two arrays and then immediately flatten it into one array? For ordering?

Reply:
I believe this orders the tags in a way, which allows to see "late" tags first (if they exceed set frequency by late reporting).

Reply:
I see it now - late tags are marked and then the partition takes place, which puts late tags into first array and rest to the second. Then they are flattened